internal/client/githubclient/githubclient.go

```diff
@@ -0,0 +1,161 @@
+//
+// Copyright 2023 The GUAC Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package githubclient
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/google/go-github/github"
+	"github.com/guacsec/guac/internal/client"
+	"golang.org/x/oauth2"
+)
+
+// Wrapper interface for Github
+type GithubClient interface {
+	// GetLatestRelease fetches the latest release for a repo
+	GetLatestRelease(ctx context.Context, owner string, repo string) (*client.Release, error)
+
+	// GetCommitSHA1 fetches the commit SHA in a repo based on a tag, branch head, or other ref.
+	// NOTE: Github release 2022-11-28 and similar server returns a commitish for a release.
+	// The release commitish can be a commit, branch name, or a tag.
+	// We need to resolve it to a commit.
+	GetCommitSHA1(ctx context.Context, owner string, repo string, ref string) (string, error)
+
+	// GetReleaseByTagSlices fetches metadata regarding releases for a given tag. If the tag is the empty string,
+	// it should just return the latest.
+	GetReleaseByTag(ctx context.Context, owner string, repo string, tag string) (*client.Release, error)
+
+	// GetReleaseAsset fetches the content of a release asset, e.g. artifacts, metadata documents, etc.
+	GetReleaseAsset(asset client.ReleaseAsset) (*client.ReleaseAssetContent, error)
+}
+
+type githubClient struct {
+	ghClient   *github.Client
+	httpClient *http.Client
+}
+
+var _ GithubClient = &githubClient{}
```

Comment:
This is just short hand to ensure that `githubClient` in fact implements the `GithubClient` interface as it will fail to compile if something changes.